### PR TITLE
RetroAchievements: Try to eliminate possible race conditions on load

### PR DESCRIPTION
**Pull request by hrydgard**
Should fix achievements from frontends.

Was not able to reproduce the problem consistently, so doing two different fixes here.

Part of #17631

### CODE REVIEW

File: Core/RetroAchievements.cpp [hunk 1]
```diff
@@ -630,6 +635,13 @@ struct FileContext {
 static BlockDevice *g_blockDevice;
 
 void SetGame(const Path &path, FileLoader *fileLoader) {
+	// If we are currently logging in, give it a couple of seconds. This is not ideal, but works.
```
Comment (unknownbrackets) [inline Core/RetroAchievements.cpp:638]: Is it just that we need to block until the login is complete?  Sounds like a job for a promise, rather.  Seems like it wouldn't be hard to cond var on the login either.

If it has a popular day 3s might not be enough.  Or if it gets some backend upgrades maybe that's slowing down game startup.

-[Unknown]

Comment (hrydgard) [reply Core/RetroAchievements.cpp:638]: Yeah, I was just lazy and wanted a quick fix ..  Condvar on the login is probably the proper way to go. 

Timeouts in general are a tricky area.. I don't think we have a good solution for that in our HTTP client?

Comment (unknownbrackets) [reply Core/RetroAchievements.cpp:638]: It does have a mechanism for timeouts and they do work (used for e.g. when you have recent remote discs and you're away from your home network), and also cancellation.

See `client.SetDataTimeout()`.  Also Connect or something has a timeout param.

-[Unknown]

Comment (hrydgard) [reply Core/RetroAchievements.cpp:638]: Hm, I'll take a look. I think I just looked at the Download object (which is what you get in those callbacks), which doesn't have a .TimedOut() or anything.

**Pull request merged**